### PR TITLE
[fronius] Add additional Inverter DC channels

### LINKED ISSUE #12856: [Fronius] Add channels for secondary and tertiary DC input
The channels of the inverter thing for Fronius include DC current and DC voltage. Unfortunately, those channels just get the readings for the first DC input of the inverter. Readings **for the second DC input as well as for the battery input are not yet parsed**:

image

When I check for the json provided by the `solar_api`:

`http://<inverter-ip>/solar_api/v1/GetInverterRealtimeData.cgi?Scope=Device&DeviceId=1&DataCollection=CommonInverterData`

I get this response json:

```
{
   "Body" : {
      "Data" : {
         "DAY_ENERGY" : {
            "Unit" : "Wh",
            "Value" : null
         },
         "DeviceStatus" : {
            "ErrorCode" : 0,
            "InverterState" : "Running",
            "StatusCode" : 7
         },
         "FAC" : {
            "Unit" : "Hz",
            "Value" : 49.982288360595703
         },
         "IAC" : {
            "Unit" : "A",
            "Value" : 2.8039026260375977
         },
         "IDC" : {
            "Unit" : "A",
            "Value" : 0.025438997894525528
         },
         "IDC_2" : {
            "Unit" : "A",
            "Value" : 1.4097353219985962
         },
         "IDC_3" : {
            "Unit" : "A",
            "Value" : null
         },
         "PAC" : {
            "Unit" : "W",
            "Value" : 660.802978515625
         },
         "SAC" : {
            "Unit" : "VA",
            "Value" : 660.84747314453125
         },
         "TOTAL_ENERGY" : {
            "Unit" : "Wh",
            "Value" : 52214.948611111111
         },
         "UAC" : {
            "Unit" : "V",
            "Value" : 236.84129333496094
         },
         "UDC" : {
            "Unit" : "V",
            "Value" : 422.36505126953125
         },
         "UDC_2" : {
            "Unit" : "V",
            "Value" : 484.74273681640625
         },
         "UDC_3" : {
            "Unit" : "V",
            "Value" : null
         },
         "YEAR_ENERGY" : {
            "Unit" : "Wh",
            "Value" : null
         }
      }
   },
   "Head" : {
      "RequestArguments" : {
         "DataCollection" : "CommonInverterData",
         "DeviceId" : "1",
         "Scope" : "Device"
      },
      "Status" : {
         "Code" : 0,
         "Reason" : "",
         "UserMessage" : ""
      },
      "Timestamp" : "2022-06-01T17:36:17+00:00"
   }
}
```

The measurements for the second DC input are revealed in the `UDC_2 `and `IDC_2 `stanza.
I'd assume, that the batteries input values will go into `UDC_3 `and `IDC_3`. Values are still `null` here as the battery is not connected currently.

@jimtng Could you parse the second and third DC input too and provide them as channels in the binding?

_Btw.: The inverters web interface directly shows values for the DC power for each input. Sure, it's simple to calculate the power from the current and voltage given, but if the power of the inputs would be available directly as channels, it would save the effort for additional rules doing the math. Means, **I'd love to get DC power per input as channel** too ;-)_

Let me know, if I can provide more information or if there is something I can test on my side.

Many thanks upfront,
Curlyel

```
openHAB: 3.2.0 release build
Fronius:  Symo Gen24 10.0 Version 1.19.7-1

```

### PR DESCRIPTION
**Pull request by jimtng**
Resolve #12856

Add the following new channels for Fronius Gen24 inverters:
* inverterdatachannelidc2
* inverterdatachannelidc3
* inverterdatachanneludc2
* inverterdatachanneludc3

Add calculated DC power channels:
* inverterdatachannelpdc
* inverterdatachannelpdc2
* inverterdatachannelpdc3

### CODE REVIEW

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -202,6 +218,18 @@
...
 	</channel-type>
+	<channel-type id="udc2">
+		<item-type>Number:ElectricPotential</item-type>
+		<label>DC Voltage 2</label>
+		<description>DC voltage</description>
+		<state pattern="%.1f V" readOnly="true"></state>
+	</channel-type>
+	<channel-type id="udc3">
+		<item-type>Number:ElectricPotential</item-type>
+		<label>DC Voltage</label>
```
Comment (jlaur) [inline bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml:229]: As already noticed by @curlyel:
```suggestion
		<label>DC Voltage 3</label>
```

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -202,6 +218,18 @@
...
+		<description>DC voltage 3</description>
+		<state pattern="%.1f V" readOnly="true"></state>
+	</channel-type>
```
Comment (jlaur) [inline bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml:232]: i18n properties file needs to be regenerated.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Some early feedback for your draft.

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -202,6 +218,18 @@
 		<description>DC voltage</description>
 		<state pattern="%.1f V" readOnly="true"></state>
 	</channel-type>
+	<channel-type id="udc2">
+		<item-type>Number:ElectricPotential</item-type>
+		<label>DC Voltage 2</label>
+		<description>DC voltage</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml:224]:
```suggestion
		<description>DC voltage 2</description>
```

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (jimtng) [reply bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml]: Thanks for spotting this. I have updated the label.

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml (hunk 3 above)
Comment (jimtng) [reply bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml]: Thanks for spotting this. I have updated the description.

File: bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (jimtng) [reply bundles/org.openhab.binding.fronius/src/main/resources/OH-INF/thing/thing-types.xml:232]: Thanks, I forgot. I've regenerated the i18n file now.

**Review by jlaur — APPROVED** [review]
LGTM - thanks!

**Pull request merged**